ClientJmxMetricsTest.testNoMBeanLeak

_master_ (commit 320b492c89c9c32403de6798d7f4effadb53b3c4)

Failed on ZuluJDK10 : http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.x-ZuluJDK10/54/testReport/com.hazelcast.client.internal.metrics/ClientJmxMetricsTest/testNoMBeanLeak/

**Stacktrace**
```
java.lang.AssertionError: expected:<0> but was:<7>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.internal.metrics.jmx.JmxPublisherTestHelper.assertNoMBeans(JmxPublisherTestHelper.java:50)
	at com.hazelcast.client.internal.metrics.ClientJmxMetricsTest.testNoMBeanLeak(ClientJmxMetricsTest.java:52)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

Might be related with https://github.com/hazelcast/hazelcast/commit/8f3e2c16a743a009a7c6f2a9389fc36427a424c6 ? @blazember fyi.

--- a/hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleAppTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/console/ClientConsoleAppTest.java
@@ -16,10 +16,9 @@
 
 package com.hazelcast.client.console;
 
-import com.hazelcast.client.HazelcastClient;
 import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.test.TestHazelcastFactory;
 import com.hazelcast.config.Config;
-import com.hazelcast.core.Hazelcast;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.map.IMap;
 import com.hazelcast.test.HazelcastSerialClassRunner;
@@ -52,7 +51,8 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
     private static ByteArrayOutputStream baos;
     private static PrintStream printStream;
 
-    private  HazelcastInstance hazelcastInstance;
+    private HazelcastInstance hazelcastInstance;
+    private TestHazelcastFactory hazelcastFactory = new TestHazelcastFactory();
 
     @BeforeClass
     public static void beforeClass() {
@@ -67,17 +67,18 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
     @Before
     public void before() {
         resetSystemOut();
-        hazelcastInstance = Hazelcast.newHazelcastInstance(new Config());
+        hazelcastInstance = hazelcastFactory.newHazelcastInstance(new Config());
     }
 
     @After
     public void after() {
         hazelcastInstance.shutdown();
+        hazelcastFactory.terminateAll();
     }
 
     @Test
     public void executeOnKey() {
-        ClientConsoleApp consoleApp = new ClientConsoleApp(HazelcastClient.newHazelcastClient(new ClientConfig()), printStream);
+        ClientConsoleApp consoleApp = new ClientConsoleApp(hazelcastFactory.newHazelcastClient(new ClientConfig()), printStream);
         for (int i = 0; i < 100; i++) {
             consoleApp.handleCommand(String.format("executeOnKey message%d key%d", i, i));
             assertTextInSystemOut("message" + i);
@@ -89,7 +90,7 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
      */
     @Test
     public void mapPut() {
-        HazelcastInstance hz = HazelcastClient.newHazelcastClient(new ClientConfig());
+        HazelcastInstance hz = hazelcastFactory.newHazelcastClient(new ClientConfig());
         ClientConsoleApp consoleApp = new ClientConsoleApp(hz, printStream);
 
         IMap<String, String> map = hz.getMap("default");
@@ -113,7 +114,7 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
      */
     @Test
     public void mapRemove() {
-        HazelcastInstance hz = HazelcastClient.newHazelcastClient(new ClientConfig());
+        HazelcastInstance hz = hazelcastFactory.newHazelcastClient(new ClientConfig());
         ClientConsoleApp consoleApp = new ClientConsoleApp(hz, printStream);
         IMap<String, String> map = hz.getMap("default");
         map.put("a", "valueOfA");
@@ -130,7 +131,7 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
      */
     @Test
     public void mapDelete() {
-        HazelcastInstance hz = HazelcastClient.newHazelcastClient(new ClientConfig());
+        HazelcastInstance hz = hazelcastFactory.newHazelcastClient(new ClientConfig());
         ClientConsoleApp consoleApp = new ClientConsoleApp(hz, printStream);
         IMap<String, String> map = hz.getMap("default");
         map.put("a", "valueOfA");
@@ -147,7 +148,7 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
      */
     @Test
     public void mapGet() {
-        HazelcastInstance hz = HazelcastClient.newHazelcastClient(new ClientConfig());
+        HazelcastInstance hz = hazelcastFactory.newHazelcastClient(new ClientConfig());
         ClientConsoleApp consoleApp = new ClientConsoleApp(hz, printStream);
         hz.<String, String>getMap("default").put("testGetKey", "testGetValue");
         consoleApp.handleCommand("m.get testGetKey");
@@ -159,7 +160,7 @@ public class ClientConsoleAppTest extends HazelcastTestSupport {
      */
     @Test
     public void mapPutMany() {
-        HazelcastInstance hz = HazelcastClient.newHazelcastClient(new ClientConfig());
+        HazelcastInstance hz = hazelcastFactory.newHazelcastClient(new ClientConfig());
         ClientConsoleApp consoleApp = new ClientConsoleApp(hz, printStream);
         IMap<String, ?> map = hz.getMap("default");
         consoleApp.handleCommand("m.putmany 100 8 1000");

